src/platform/graphics/blend-state.js

```diff
@@ -0,0 +1,222 @@
+        this.blend = blend;
+    }
+
+    /**
+     * Enables or disables blending.
+     *
+     * @type {boolean}
+     */
+    set blend(value) {
+        this.target0 = BitPacking.set(this.target0, value ? 1 : 0, blendShift);
+    }
+
+    get blend() {
+        return BitPacking.all(this.target0, blendShift);
+    }
+
+    setColorBlend(op, srcFactor, dstFactor) {
+        this.target0 = BitPacking.set(this.target0, op, colorOpShift, opMask);
+        this.target0 = BitPacking.set(this.target0, srcFactor, colorSrcFactorShift, factorMask);
+        this.target0 = BitPacking.set(this.target0, dstFactor, colorDstFactorShift, factorMask);
+    }
+
+    setAlphaBlend(op, srcFactor, dstFactor) {
+        this.target0 = BitPacking.set(this.target0, op, alphaOpShift, opMask);
+        this.target0 = BitPacking.set(this.target0, srcFactor, alphaSrcFactorShift, factorMask);
+        this.target0 = BitPacking.set(this.target0, dstFactor, alphaDstFactorShift, factorMask);
+    }
+
+    setColorWrite(redWrite, greenWrite, blueWrite, alphaWrite) {
+        this.redWrite = redWrite;
+        this.greenWrite = greenWrite;
+        this.blueWrite = blueWrite;
+        this.alphaWrite = alphaWrite;
+    }
+
+    get colorOp() {
+        return BitPacking.get(this.target0, colorOpShift, opMask);
+    }
+
+    get colorSrcFactor() {
+        return BitPacking.get(this.target0, colorSrcFactorShift, factorMask);
+    }
+
+    get colorDstFactor() {
+        return BitPacking.get(this.target0, colorDstFactorShift, factorMask);
+    }
+
+    get alphaOp() {
+        return BitPacking.get(this.target0, alphaOpShift, opMask);
+    }
+
+    get alphaSrcFactor() {
+        return BitPacking.get(this.target0, alphaSrcFactorShift, factorMask);
+    }
+
+    get alphaDstFactor() {
+        return BitPacking.get(this.target0, alphaDstFactorShift, factorMask);
+    }
+
+    set redWrite(value) {
+        this.target0 = BitPacking.set(this.target0, value ? 1 : 0, redWriteShift);
+    }
+
+    get redWrite() {
+        return BitPacking.all(this.target0, redWriteShift);
+    }
+
+    set greenWrite(value) {
+        this.target0 = BitPacking.set(this.target0, value ? 1 : 0, greenWriteShift);
+    }
+
+    get greenWrite() {
+        return BitPacking.all(this.target0, greenWriteShift);
+    }
+
+    set blueWrite(value) {
+        this.target0 = BitPacking.set(this.target0, value ? 1 : 0, blueWriteShift);
+    }
+
+    get blueWrite() {
+        return BitPacking.all(this.target0, blueWriteShift);
+    }
+
+    set alphaWrite(value) {
+        this.target0 = BitPacking.set(this.target0, value ? 1 : 0, alphaWriteShift);
+    }
+
+    get alphaWrite() {
+        return BitPacking.all(this.target0, alphaWriteShift);
+    }
+
+    get allWrite() {
+        // return a number with all 4 bits, for fast compare
+        return BitPacking.get(this.target0, allWriteShift, allWriteMasks);
+    }
+
+    copy(rhs) {
```

Comment:
Would a `clone` be useful too? We seem to be developing a `copy` + `clone` interface across many API classes.

Reply:
It might be useful, so this has been added now.